File: C#/6-kyu/Counting Duplicates.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Kata
{
  public static int DuplicateCount(string str)
  {
    string lowerInput = str.ToLower();

    var characterCounts = new Dictionary<char, int>();
    
    foreach (char character in lowerInput) {
      if (characterCounts.ContainsKey(character)) {
        characterCounts[character]++;
      } else {
        characterCounts[character] = 1;
      }
    }
    
    return characterCounts.Count(kvp => kvp.Value > 1);
  }
}

File: Go/6-kyu/Counting Duplicates.go
package kata

import "strings"

func duplicate_count(s1 string) int {
  s1 = strings.ToLower(s1)
  
  countMap := make(map[rune]int)
  
  for _, char := range s1 {
    countMap[char]++
  }
  
  duplicateCount := 0
  
  for _, count := range countMap {
    if count > 1 {
      duplicateCount++
    }
  }
  
  return duplicateCount
}
